dd-sdk-android/src/test/kotlin/com/datadog/android/tracing/internal/TracerTest.kt

```diff
@@ -0,0 +1,76 @@
+package com.datadog.android.tracing.internal
+
+import android.app.Application
+import com.datadog.android.Datadog
+import com.datadog.android.tracing.Tracer
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.android.utils.mockContext
+import com.datadog.tools.unit.invokeMethod
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+
+@MockitoSettings(strictness = Strictness.STRICT_STUBS)
+@ForgeConfiguration(Configurator::class)
+internal class TracerTest {
+
+    lateinit var underTest: Tracer.Builder
+    lateinit var mockAppContext: Application
+    lateinit var fakeToken: String
+    lateinit var fakeServiceName: String
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        fakeServiceName = forge.anAlphabeticalString()
+        fakeToken = forge.anHexadecimalString()
+        mockAppContext = mockContext()
+        Datadog.initialize(mockAppContext, fakeToken)
+    }
+
+    @AfterEach
+    fun `tear down`() {
+        Datadog.invokeMethod("stop")
+    }
+
+    @Test
+    fun `it will build a valid Tracer`(forge: Forge) {
+        // given
+        underTest = Tracer.Builder()
+
+        // when
+        val tracer = underTest.setServiceName(fakeServiceName).build()
+
+        // then
+        assertThat(tracer).isNotNull()
+        val span = tracer.buildSpan(forge.anAlphabeticalString()).start()
```

Comment:
Why do we have to build the span in this test? Shouldn't we also somehow assert on this?